File: app/src/main/java/com/tuvakov/zetube/android/dagger/AppComponent.java
package com.tuvakov.zetube.android.dagger;

import com.tuvakov.zetube.android.VideoFeedSyncService;
import com.tuvakov.zetube.android.ui.feed.VideoFeedActivity;
import com.tuvakov.zetube.android.ui.player.PlayerActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, RoomModule.class, ExecutorModule.class})
public interface AppComponent {

    void injectVideoFeedActivityFields(VideoFeedActivity videoFeedActivity);

    void injectPlayerActivityFields(PlayerActivity playerActivity);

    void injectVideoFeedSyncService(VideoFeedSyncService service);

    @Component.Factory
    interface Factory {
        AppComponent create(AppModule appModule);
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/dagger/AppComponent.kt
package com.tuvakov.zetube.android.dagger

import com.tuvakov.zetube.android.ui.channels.ChannelsActivity
import com.tuvakov.zetube.android.ui.feed.VideoFeedActivity
import com.tuvakov.zetube.android.ui.player.PlayerActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, RoomModule::class])
interface AppComponent {
    fun injectVideoFeedActivityFields(videoFeedActivity: VideoFeedActivity)
    fun injectPlayerActivityFields(playerActivity: PlayerActivity)
    fun injectChannelsActivity(channelsActivity: ChannelsActivity)

    @Component.Factory
    interface Factory {
        fun create(appModule: AppModule): AppComponent
    }
}